cmd/cloudshell_open/main.go

```diff
@@ -274,12 +279,16 @@ func run(c *cli.Context) error {
 	cmdColor.Println("\\")
 	cmdColor.Printf("\t  --memory=%s", parameter(defaultRunMemory))
 	cmdColor.Println("\\")
-	cmdColor.Printf("\t  --allow-unauthenticated\n")
+	if requireAuth {
+		cmdColor.Printf("\t  --no-allow-unauthenticated\n")
+	} else {
+		cmdColor.Printf("\t  --allow-unauthenticated\n")
+	}
 
 	end = logProgress(fmt.Sprintf("Deploying service %s to Cloud Run...", serviceLabel),
 		fmt.Sprintf("Successfully deployed service %s to Cloud Run.", serviceLabel),
 		"Failed deploying the application to Cloud Run.")
-	url, err := deploy(project, serviceName, image, region, envs)
+	url, err := deploy(project, serviceName, image, region, envs, requireAuth)
```

Comment:
aside: I feel like we'll soon have to introduce a DeployOptions property-bag here :)